src/main/java/org/springframework/data/couchbase/repository/query/N1qlQueryCreator.java

```diff
@@ -69,7 +72,13 @@ public N1qlQueryCreator(final PartTree tree, final ParameterAccessor accessor, f
 		this.queryMethod = queryMethod;
 		this.converter = converter;
 		this.bucketName = bucketName;
-		this.entity = converter.getMappingContext().getPersistentEntity(queryMethod.getReturnedObjectType());
+
+		Class<?> returnedObjectType = queryMethod.getReturnedObjectType();
+		if (ClassUtils.isPrimitiveOrWrapper(returnedObjectType)) {
+			this.entity = new BasicCouchbasePersistentEntity(ClassTypeInformation.from(returnedObjectType));
```

Comment:
Entities should not be generally created for return types. A query is always contextually related to an entity, but using the return type doesn't match that assumption. You could return `void` or `Integer` and entity creation inspects constructors etc. That could should be adopted properly to detect whether the return type qualifies as entity.

Reply:
This was specifically added to handle methods of the type  Long countByxxxx()- the existing code was failing following the the jdk16 enforcement of InaccessibleObjectException.  The code here corresponds to https://github.com/spring-projects/spring-data-mongodb/blob/7a640256698f0772c1509c58e4bc71652a448567/spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/query/MongoQueryMethod.java#L143

I'll see if there is a better way to handle this - at least factor out the common code that is repeated in multiple places.

Reply:
yuck! I see, the code in MongoDB isn't even much better although it uses the `EntityMetadata` approach. Creating `PersistentEntity` objects can easily lead to `InaccessibleObjectException` and my comment above is motivated by using code that isn't likely to lead to `InaccessibleObjectException`.

Reply:
The cases where I had made changes were hitting exceptions because of a latent bug - they were using the returnType of the method where they should have been using the entityType. After fixing that bug, the subsequent getPersistentEntity( type ) always returns an existing entity.  I think we are ok here.